File: Anole/HTMLParser.cs
using System;
using System.Xml;
using System.Windows.Forms;

namespace Anole
{
	/// <summary>
	/// Summary description for HTMLParser.
	/// </summary>
	public class HTMLParser
	{

		XmlDocument myDocument;
	
		public HTMLParser()
		{
			myDocument=new XmlDocument();
		}

		public void Parse(HTMLDocument document)
		{
			try
			{
				string s=document.HTML.Replace("<br>","<br/>");
				s=s.Replace("<hr>","<hr/>");
				myDocument.LoadXml(s);
				document.RootElement=new HTMLElement(ref document, null, myDocument.FirstChild);
			}
			catch
			{
				
			}
		}
	}
}

File: Anole/HTMLParser.vb
Imports System.Xml

Namespace Anole

    Public Class HTMLParser

        Private _Document As New XmlDocument

        Public Sub Parse(ByVal Document As Anole.HTMLDocument)
            Try
                Dim s As String = Document.HTML.Replace("<br>", "<br/>").Replace("<hr>", "<hr/>")
                _Document.LoadXml(s)

            Catch ex As Exception

            End Try
        End Sub

    End Class

End Namespace